### PR TITLE
camera db recog integration

### LINKED ISSUE #63: Pull cameras from db
In RecogScript, pull camera IPs from db instead of from text file

### LINKED ISSUE #64: Fair share between camera frames
In RecogScript, we want to pull frames from camera ip’s evenly

### PR DESCRIPTION
**Pull request by katebrune**
- recog script now pulls camera feeds from db
- recog script is now just main, moved process_video_to_encode & check_encodings to FacialRecog.py
  - added function get_frame_from_camera, takes a camera object from db and returns a frame
  - changed name process_video_to_encode -> compare_frame_and_encoding, this made sense because the function was just comparing the frame and encoding to get a list of comparisons, rather than processing anything
  - changed name check_encodings -> process_comparisons, this made sense because we aren't checking encodings, we are checking the comparison output

- updated api routes to be compatible with new camera db
  - added 'camera_id' field to seen db
  - added 'camera_id' field to alerts db
  - when adding a new seen, now you pass the ref_id and camera_id
  - when adding a new alert, now you pass the ref_id and camera_id
  - updated get_seen_time_interval to accept a camera_id, so now it returns all seen in a time interval for a specific camera
  - updated get_alerts_time_interval to accept a camera_id, so now it returns all alerts in a time interval for a specific camera

- found a bug with process_comparisons (formerly check_encodings) logic:
 previously: 
image

the for loop is going over each comparison, and it was saying if true, mark seen, alert, which is all good. but if false, add this image as an unknown person. but consider how this is working. If me and ryan have encodings in the system, and i'm the only one in camera, then the comparison data will look like [[True], [False]]. so it would mark me as seen (okay), but then when it went through for ryan and noted i wasn't him, it would add this frame of me to system as unknown, which isn't right.

updated:
image

face_recognized is a list that is the length of the amount of faces in the frame (the same length of each comparison array), and it is initialized to be all false. 

we keep the code the same for the beginning basically, iterating through each comparison list and if a true is found, reporting the person and seen and checking for an alert. additionally, we mark the index of face_recognized to be true, because that face in the frame has been seen.

then, once all comparison lists are checked, we iterate through face_recognized to see if there were any faces in the frame that weren't recognized. only then do we add a new unknown person to the system.

closes #63, closes #64

### CODE REVIEW

File: FacialRecog.py [hunk 1]
```diff
@@ -148,6 +171,64 @@ def compare_encodings(frame_encodings, all_encodings):
...
+
+"""
+given a list of comparisons, list of ids, camera, frame,
+processes comparisons
+"""
+
+
+def process_comparisons(comparisons, ids, camera, frame):
+    if comparisons is not None:
+        print(comparisons)
```
Comment (rgoluch) [inline FacialRecog.py:215]: We might want to delete the print statements
```suggestion
```

File: FacialRecog.py [hunk 2]
```diff
@@ -148,6 +171,64 @@ def compare_encodings(frame_encodings, all_encodings):
...
+        for comparison in comparisons:
+            if True in comparison[:len(comparison)]:
+                face_recognized[comparison.index(True)] = True
+                id = ids[comparisons.index(comparison)]
+                add_new_seen(id, str(camera['_id']))
+                check_for_alert(id, str(camera['_id']))
+
+        unknown_encodings, unknown_images = get_images_and_encodings(frame)
+        i = 0
+        print(face_recognized)
```
Comment (rgoluch) [inline FacialRecog.py:226]:
```suggestion
```

**Review by rgoluch — COMMENTED** [review]
Looks good to me outside of the print statements. Good catch with the double counting of true/false!

**Review by rgoluch — CHANGES_REQUESTED** [review]
The print statements from my previous comment

File: FacialRecog.py (hunk 1 above)
Comment (katebrune) [reply FacialRecog.py:215]: just left them in there so if yall ran it you would see what i meant, but ya we can remove them

**Pull request merged**